### PR TITLE
[Feat] Throttle button onClick events

### PR DESCRIPTION
**Pull request by taufiqmmhd**
This PR throttles all button onClick events that invoke API calls so that end-users will not send multiple API calls if they choose to spam click those buttons

### CODE REVIEW

File: src/components/ImageSettingsModal.jsx [hunk 1]
```diff
@@ -103,8 +104,8 @@ export default class ImageSettingsModal extends Component {
               />
             </div>
             <div className={elementStyles.modalButtons}>
-              <button type="button" className={errorMessage ? elementStyles.disabled : elementStyles.blue} disabled={!!errorMessage} onClick={this.renameImage}>Save</button>
-              <button type="button" className={sha ? elementStyles.warning : elementStyles.disabled} onClick={this.deleteImage} disabled={!sha}>Delete</button>
+              <button type="button" className={errorMessage ? elementStyles.disabled : elementStyles.blue} disabled={!!errorMessage} onClick={_.throttle(this.renameImage, 2000)}>Save</button>
```
Comment (prestonlimlianjie) [inline src/components/ImageSettingsModal.jsx]: Also I know this won't be relevant if we decide against using throttle, but a note on style: we should define magic numbers like `2000` as a constant at the top of the file and re-use it across the file.

**Comment by kwajiehao** [comment]
> Actually I was thinking about it, throttling at 2000ms doesn't guarantee that the user can't make the same call twice if the first call takes longer than 2000ms.
> 
> I think a better solution is to set the state of the button to disabled, show some feedback message in the button like `Saving` or `Deleting` or `Updating`, and only enable the button again after the various handlers are done running.
> 
> What are your thoughts @kwajiehao @taufiqmmhd ?

Yeah that sounds good to me

File: src/components/ImageSettingsModal.jsx (hunk 1 above)
Comment (taufiqmmhd) [reply src/components/ImageSettingsModal.jsx]: yea definitely

**Comment by taufiqmmhd** [comment]
> Actually I was thinking about it, throttling at 2000ms doesn't guarantee that the user can't make the same call twice if the first call takes longer than 2000ms.
> 
> I think a better solution is to set the state of the button to disabled, show some feedback message in the button like `Saving` or `Deleting` or `Updating`, and only enable the button again after the various handlers are done running.
> 
> What are your thoughts @kwajiehao @taufiqmmhd ?

Yeah let's do that

**Comment by taufiqmmhd** [comment]
This is a horrible PR.

**Pull request closed without merging**